src/SIM/SIMinput.C

```diff
@@ -1170,11 +1171,21 @@ bool SIMinput::refine (const LR::RefineData& prm,
     }
   }
   for (size_t i = 0; i < myModel.size(); i++)
-    for(int j : conformingIndicies[i])
+  {
+    pch = dynamic_cast<ASMunstruct*>(myModel[i]);
+    IntVec secondary = pch->getOverlappingNodes(conformingIndicies[i]);
+    for(int j : conformingIndicies[i]) // add refinement from neighbours
       refineIndices[i].push_back(j);
-
+    for(int j : secondary)             // add depth refinement to make it conforming
+      refineIndices[i].push_back(j);
+  }
   for (size_t i = 0; i < myModel.size(); i++)
   {
+    // make unique list of refinement indices
+    std::sort(refineIndices[i].begin(), refineIndices[i].end());
+    auto last = std::unique(refineIndices[i].begin(), refineIndices[i].end());
+    refineIndices[i].erase(last, refineIndices[i].end());
+
```

Comment:
Her også, vurder å bruke `std::set<int>`i stedet.

Reply:
Innser nå at mye av det jeg kommenterte på i første commit er endret på i denne. Da burde heller de to committene vært squashet til en for å gjøre review-prosessen enklere og ryddigere

Reply:
Hvis du klikker på "Files changed" istedenfor "commits" øverst, så får du diff'en til hele pull requesten. Essensielt det du spør om: slik ting hadde sett ut hvis alt hadde blitt most til en enkelt commit.

Jeg tror ikke review hadde blitt enklere med få commits, tvert imot da man mister commit messages og en del logikk til hvorfor ting blir som de blir. Jeg kan derimot se argumentet med at man ikke vil ha for mange intermediate states i historikken der programmet dumper ut unødvendig mye debug print (som her hos meg) eller værst av alt: ikke kompilerer (skal ikke være tilfellet her). Hvis det fortsatt er ønskelig, så moser jeg alt sammen til en stor commit.

Reply:
For all del, alt trenger ikke å være en stor commit, men de committene som er der bør vær logisk inndelt så langt det går. Ihvertfall trengs ikke fjernings av debug print, dokumentasjons-endringer, etc. være separat. De kan med fordel inkluderes i de forrige der de hører best hjemme.